### DISCUSSION TITLE
我的PaddleOCR实现效果不如官方Demo，如何改进？

### DISCUSSION THREAD

**Discussion by AshtonChan11** [PaddlePaddle/PaddleOCR#13443] [Q&A] [answered]
你好，以下是我的测试代码：
> ocr = PaddleOCR(use_gpu=False, use_angle_cls=True, lang="ch", 
    det_model_dir="./model/ch_PP-OCRv4_det_infer", 
    rec_model_dir="./model/ch_PP-OCRv4_rec_infer", 
    cls_model_dir="./model/ch_ppocr_mobile_v2.0_cls_infer")
img_path = './screenshots/screenshot.png'
result = ocr.ocr(img_path, cls=True)

以下是我的调试结果：

> stdout: [2024/07/21 13:05:38] ppocr DEBUG: Namespace(help='==SUPPRESS==', use_gpu=False, use_xpu=False, use_npu=False, use_mlu=False, ir_optim=True, use_tensorrt=False, min_subgraph_size=15, precision='fp32', gpu_mem=500, gpu_id=0, image_dir=None, page_num=0, det_algorithm='DB', det_model_dir='./model/ch_PP-OCRv4_det_infer', det_limit_side_len=960, det_limit_type='max', det_box_type='quad', det_db_thresh=0.3, det_db_box_thresh=0.6, det_db_unclip_ratio=1.5, max_batch_size=10, use_dilation=False, det_db_score_mode='fast', det_east_score_thresh=0.8, det_east_cover_thresh=0.1, det_east_nms_thresh=0.2, det_sast_score_thresh=0.5, det_sast_nms_thresh=0.2, det_pse_thresh=0, det_pse_box_thresh=0.85, det_pse_min_area=16, det_pse_scale=1, scales=[8, 16, 32], alpha=1.0, beta=1.0, fourier_degree=5, rec_algorithm='SVTR_LCNet', rec_model_dir='./model/ch_PP-OCRv4_rec_infer', rec_image_inverse=True, rec_image_shape='3, 48, 320', rec_batch_num=6, max_text_length=25, rec_char_dict_path='D:\\anaconda3\\lib\\site-packages\\paddleocr\\ppocr\\utils\\ppocr_keys_v1.txt', use_space_char=True, vis_font_path='./doc/fonts/simfang.ttf', drop_score=0.5, e2e_algorithm='PGNet', e2e_model_dir=None, e2e_limit_side_len=768, e2e_limit_type='max', e2e_pgnet_score_thresh=0.5, e2e_char_dict_path='./ppocr/utils/ic15_dict.txt', e2e_pgnet_valid_set='totaltext', e2e_pgnet_mode='fast', use_angle_cls=True, cls_model_dir='./model/ch_ppocr_mobile_v2.0_cls_infer', cls_image_shape='3, 48, 192', label_list=['0', '180'], cls_batch_num=6, cls_thresh=0.9, enable_mkldnn=False, cpu_threads=10, use_pdserving=False, warmup=False, sr_model_dir=None, sr_image_shape='3, 32, 128', sr_batch_num=1, draw_img_save_dir='./inference_results', save_crop_res=False, crop_res_save_dir='./output', use_mp=False, total_process_num=1, process_id=0, benchmark=False, save_log_path='./log_output/', show_log=True, use_onnx=False, return_word_box=False, output='./output', table_max_len=488, table_algorithm='TableAttn', table_model_dir=None, merge_no_span_structure=True, table_char_dict_path=None, layout_model_dir=None, layout_dict_path=None, layout_score_threshold=0.5, layout_nms_threshold=0.5, kie_algorithm='LayoutXLM', ser_model_dir=None, re_model_dir=None, use_visual_backbone=True, ser_dict_path='../train_data/XFUND/class_list_xfun.txt', ocr_order_method=None, mode='structure', image_orientation=False, layout=True, table=True, ocr=True, recovery=False, use_pdf2docx_api=False, invert=False, binarize=False, alphacolor=(255, 255, 255), lang='ch', det=True, rec=True, type='ocr', savefile=False, ocr_version='PP-OCRv4', structure_version='PP-StructureV2')

现在有个问题就是效果远不如官方demo的效果

![屏幕截图 2024-07-21 132328](https://github.com/user-attachments/assets/b8cce851-a766-4491-94e3-2ebfc41fe4ea)
![屏幕截图 2024-07-21 132457](https://github.com/user-attachments/assets/7acbf922-354d-4665-a914-d007428a8fe4)

希望大家能帮我看看问题出在哪里，或提供一些调试和优化的建议。非常感谢！

**Comment by GreatV** [comment]
demo里用的是高精度模型 https://github.com/PaddlePaddle/PaddleOCR/blob/main/doc/doc_ch/models_list.md

**Reply by AshtonChan11** [reply]
噢😯，我回去试试。
还有个问题：用这个轻量模型推理尚要差不多 4 秒，如果我上了这个高精度的感觉会更慢，还有什么办法可以提高点速度吗？

**Reply by GreatV** [reply]
可以试试换推理后端、使用gpu推理、使用onnxruntime、使用openvino、或者进行量化。

**Reply by AshtonChan11** [reply]
感谢🙏，我回去尝试一下

**Reply by AshtonChan11** [reply]
结果很尴尬，还是比不得官方的demo，不知道是不是我的图片的问题
这是高精度模型识别的结果：
![屏幕截图 2024-07-21 233808](https://github.com/user-attachments/assets/5c423944-3c1d-4887-be94-24967a630bfe)
这是轻量模型的结果：
![屏幕截图 2024-07-21 234427](https://github.com/user-attachments/assets/5da524b5-7585-40cf-84b3-d52340b1bc14)
这是官方实现的效果：
![download (3)](https://github.com/user-attachments/assets/020e6a4b-d0d2-46d3-a79f-a664e859ef23)
还是比不得，我哭死..

**Reply by UserWangZz** [reply]
使用server模型的静态图模型，然后调用tools/predict_sys.py脚本试试

**Comment by Bobholamovic** [comment, marked as answer]
你好，这是PaddleOCR 2.8的一个bug（对于较大尺寸图像强制使用滑窗推理，从而得到不符合预期的结果），目前在主分支已经修复，建议可以使用开发版本的PaddleOCR尝试～
